src/Sql/Sql/VirtualCluster/Services/AzureSqlVirtualClusterAdapter.cs

```diff
@@ -0,0 +1,116 @@
+            Communicator = new AzureSqlVirtualClusterCommunicator(Context);
+        }
+
+        /// <summary>
+        /// Gets a Virtual Cluster in a resource group
+        /// </summary>
+        /// <param name="resourceGroupName">The name of the resource group</param>
+        /// <param name="virtualClusterName">The name of the managed instance</param>
+        /// <returns>The managed instance</returns>
+        public AzureSqlVirtualClusterModel GetVirtualCluster(string resourceGroupName, string virtualClusterName)
+        {
+            var resp = Communicator.Get(resourceGroupName, virtualClusterName);
+            return CreateVirtualClusterModelFromResponse(resp);
+        }
+
+        /// <summary>
+        /// Gets a list of all virtual clusters in a subscription
+        /// </summary>
+        /// <returns>A list of all virtual clusters</returns>
+        public List<AzureSqlVirtualClusterModel> ListVirtualClusters()
+        {
+            var resp = Communicator.List();
+
+            return resp.Select(s => CreateVirtualClusterModelFromResponse(s)).ToList();
+        }
+
+        /// <summary>
+        /// Gets a list of virtual clusters in a resource group
+        /// </summary>
+        /// <param name="resourceGroupName">The name of the resource group</param>
+        /// <returns>A list of all virtual clusters</returns>
+        public List<AzureSqlVirtualClusterModel> ListVirtualClustersByResourceGroup(string resourceGroupName)
+        {
+            var resp = Communicator.ListByResourceGroup(resourceGroupName);
+            return resp.Select(s => CreateVirtualClusterModelFromResponse(s)).ToList();
+        }
+
+        /// <summary>
+        /// Deletes a virtual cluster
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group the virtual cluster is in</param>
+        /// <param name="virtualClusterName">The name of the virtual cluster to delete</param>
+        public void RemoveVirtualCluster(string resourceGroupName, string virtualClusterName)
+        {
+            Communicator.Remove(resourceGroupName, virtualClusterName);
+        }
+
+        /// <summary>
+        /// Convert a Management.Sql.Models.VirtualCluster to AzureSqlVirtualClusterModel
+        /// </summary>
+        /// <param name="resp">The management client Virtual Cluster response to convert</param>
+        /// <returns>The converted Virtual Cluster model</returns>
+        private static AzureSqlVirtualClusterModel CreateVirtualClusterModelFromResponse(Management.Sql.Models.VirtualCluster resp)
+        {
+            AzureSqlVirtualClusterModel virtualCluster = new AzureSqlVirtualClusterModel();
+
+            // Extract the resource group name from the ID.
+            // ID is in the form:
+            // /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rgName/providers/Microsoft.Sql/virtualClusters/virtualClusterName
+            string[] segments = resp.Id.Split('/');
```

Comment:
@milostod you can use the `ResourceIdentifier` class that's in our common ARM code to safely parse a provided resource id string

Reply:
Nice, thanks for the comment.